packages/core/modules/enclosure_control.py

```diff
@@ -175,17 +188,55 @@ def run(self, new_config: types.ConfigDict) -> None:
 
     # PLC.ACTORS SETTERS
 
+    def perform_camera_power_cycle(self) -> None:
+        """Performs a power cycle of the camera every day at midnight: Turn off
+        the camera power at midnight system time. Turn on the camera power two
+        minutes later.
+        
+        Exact logic: Whenever the time is between 00:00 and 00:15, and the last
+        power down was on a different day or never logged (core has been stopped
+        since last power cycle), power down the camera. Two minutes later, power
+        up the camera. This works because this function should be run at least
+        every 10 minutes (`config.general.seconds_per_core_interval` is <= 600).
+        
+        It leads to multiple power cycles if the core is restarted between 00:00
+        and 00:15, but this doesn't break any hardware. The only way to circumvent
+        this is adding it to the persistent state which is not worth it right now."""
+
+        current_time = datetime.datetime.now()
+
+        if (current_time.hour == 0) and (current_time.minute < 15):
+            if (self.last_camera_power_down_time is None) or (
+                self.last_camera_power_down_time.date() < current_time.date()
+            ):
+                self.last_camera_power_down_time = current_time
+                self.last_camera_power_up_time = None
+                self.plc_interface.set_power_camera(False)
+                logger.info("Powering down the camera.")
+
+        if (self.last_camera_power_down_time
+            is not None) and (self.last_camera_power_up_time is None):
+            if (current_time -
+                self.last_camera_power_down_time).total_seconds() > 120:
+                self.last_camera_power_up_time = current_time
+                self.plc_interface.set_power_camera(True)
+                logger.info("Powering up the camera.")
+
+        self.plc_interface.set_power_camera(False)
+
```

Comment:
This seems a bit complicated. 
You should be able to switch the power of the camera to False on a certain trigger.
In a future main-loop call you can read the PLC state for the camera power == False and turn the camera power back on.

Reply:
Do you have a proposal?

Reply:
I would go like this:
init: next_camera_shutdown = "Next day 0:01".

during main loop:
if read_camera_power = off:
set_camera_power = on

if next_camera_shutdown:
    set_camera_power = off
    next_camera_shutdown = "next day 0:01"

Reply:
And we also need to start the script to show the camera after a power cycle.